lib/tls/session.toit

```diff
@@ -180,6 +213,18 @@ class Session:
       throw "TLS_ALREADY_HANDSHAKEN"
 
     handshake_in_progress_ = monitor.Latch
+
+    if session_state:
+      toit_handshake_ = ToitHandshake_ this
+      try:
+        toit_handshake_.handshake
+        // Connected.
+        return
+      finally: | is_exception exception |
+        value := is_exception ? exception.value : null
+        handshake_in_progress_.set value
```

Comment:
The latch needs to know if you set a value or an exception, so the call to get can throw if necessary.

```suggestion
        value := is_exception ? exception.value : null
        handshake_in_progress_.set value --exception=is_exception
```